src/lib/Hydra/Controller/User.pm

```diff
@@ -37,7 +37,11 @@ sub login_POST {
         accessDenied($c, "Bad username or password.")
     }
 
-    currentUser_GET($self, $c);
+    $self->status_found(
+        $c,
+        location => $c->uri_for("current-user"),
+        entity => {}
+    );
```

Comment:
Note this is a minor change in the API: clients will need to follow the redirect to get the same response it used to. I'm inclined to think this is a non-issue, but we could replicate the old response body here if needed, by a subsequent PR.